tests/gtests/test_SW_Weather.cc

```diff
@@ -686,5 +698,315 @@ namespace {
+     }
+
+     TEST(DailyWeatherInputTest, DailyGridMet) {
+
+         /*
+            This section tests humidity-related values that can be averaged,
+            max/min relative humidity, and the calculation of actual vapor pressure,
+            based on the average relative humidity value, is calculated reasonably.
+
+            * This section uses the test directory "*_gridmet".
+          */
+
+         SW_WEATHER *w = &SW_Weather;
+         double result, expectedResult;
+         int yearIndex = 0, year = 1980;
+
+        /* Test correct priority is being given to input values from DAYMET */
+         SW_WTH_setup();
+
+             // Switch directory to gridmet input folder
+         strcpy(w->name_prefix, "Input/data_weather_gridmet/weath");
+
+             // Manually edit index/flag arrays in SW_WEATHER to make test as
+             // realistic as possible
+             // Note: Indices are based on the directory:
+             // Input/data_weather_gridmet/weath.1980
+         w->dailyInputIndices[WIND_SPEED] = 3;
+         w->dailyInputIndices[REL_HUMID_MAX] = 4;
+         w->dailyInputIndices[REL_HUMID_MIN] = 5;
+         w->dailyInputIndices[SHORT_WR] = 6;
+         w->dailyInputFlags[REL_HUMID_MAX] = swTRUE;
+         w->dailyInputFlags[REL_HUMID_MIN] = swTRUE;
+         w->dailyInputFlags[WIND_SPEED] = swTRUE;
+         w->dailyInputFlags[SHORT_WR] = swTRUE;
+         w->n_input_forcings = 7;
+
+             // Using the new inputs folder, read in year = 1980
+         _read_weather_hist(
+             year,
+             w->allHist[0],
+             w->name_prefix,
+             w->n_input_forcings,
+             w->dailyInputIndices,
+             w->dailyInputFlags
+         );
+
+         result = w->allHist[yearIndex]->r_humidity_daily[0];
+
+             // Get expected average from Input/data_weather_gridmet/weath.1980 day 1
+             // hursmax_pct and hursmin_pct
+         expectedResult = (74.17 + 31.42) / 2.;
+
+             // Test if the average of relative humidity has been calculated
+             // instead of being based on huss for the first day of 1980
+         EXPECT_NEAR(result, expectedResult, tol3);
+
+         result = w->allHist[yearIndex]->actualVaporPressure[0];
+
+             // Get expected result from Input/data_weather_gridmet/weath.1980 day 1
+             // hursmax_pct, hursmin_pct, Tmax_C, and Tmin_C
+         expectedResult = actualVaporPressure2(74.17, 31.42, -3.18, -12.32);
+
+             // Based on the max/min of relative humidity, test if actual vapor pressure
+             // was calculated reasonably
+         EXPECT_NEAR(result, expectedResult, tol6);
+
+         // Make sure calculations and set input values are within reasonable range
+         checkAllWeather(w);
+
+         // Reset directory to "data_weather"
+         strcpy(w->name_prefix, "Input/data_weather/weath");
```

Comment:
This test (and others that modify global state) need to fully reset all global states (instead of partially with `strcpy(w->name_prefix, "Input/data_weather/weath");`
```
    // Reset to previous global state
    Reset_SOILWAT2_after_UnitTest();
```